mne/viz/evoked.py

```diff
@@ -1529,162 +1686,203 @@ def plot_compare_evokeds(evokeds, picks=list(), gfp=False, colors=None,
+                        type(ci))
+
     # get and set a few limits and variables (times, channels, units)
-    example = (evokeds[conditions[0]]
-               if isinstance(evokeds[conditions[0]], Evoked)
-               else evokeds[conditions[0]][0])
-    if not isinstance(example, Evoked):
-        raise ValueError("evokeds must be an instance of mne.Evoked "
-                         "or a collection of mne.Evoked's")
-    times = example.times
+    one_evoked = evokeds[conditions[0]][0]
+    times = one_evoked.times
+    info = one_evoked.info
+    ch_names = one_evoked.ch_names
     tmin, tmax = times[0], times[-1]
-    if (tmin >= 0 or tmax <= 0) and vlines == [0.]:
-        vlines = list()
+
+    if vlines == "auto" and (tmin < 0 and tmax > 0):
+        vlines = [0.]
+    if not isinstance(vlines, (list, tuple)):
+        raise TypeError("vlines must be a list or tuple, not " + type(vlines))
 
     if isinstance(picks, Integral):
         picks = [picks]
-    elif len(picks) == 0:
-        warn("No picks, plotting the GFP ...")
+    elif picks is None:
+        logger.info("No picks, plotting the GFP ...")
         gfp = True
-        picks = _pick_data_channels(example.info)
+        picks = _pick_data_channels(info)
+
+    if not isinstance(picks, (list, np.ndarray)):
+        raise TypeError("picks should be a list or np.array of integers. "
+                        "Got %s." % type(picks))
 
-        if len(picks) == 0:
-            raise ValueError("No valid channels were found to plot the GFP. " +
-                             "Use 'picks' instead to select them manually.")
+    if len(picks) == 0:
+        raise ValueError("No valid channels were found to plot the GFP. " +
+                         "Use 'picks' instead to select them manually.")
 
     if ylim is None:
         ylim = dict()
 
     # deal with picks: infer indices and names
     if gfp is True:
-        show_sensors = False if show_sensors is None else show_sensors
+        if show_sensors is None:
+            show_sensors = False  # don't show sensors for GFP
         ch_names = ['Global Field Power']
         if len(picks) < 2:
             raise ValueError("A GFP with less than 2 channels doesn't work, "
-                             "please pick more channels.")
+                             "please pick more than %d channels." % len(picks))
+    else:
+        if show_sensors is None:
+            show_sensors = True  # show sensors when not doing GFP
+        ch_names = [one_evoked.ch_names[pick] for pick in picks]
+
+    picks_by_types = channel_indices_by_type(info, picks)
+    # keep only channel types for which there is a channel:
+    ch_types = [t for t in picks_by_types if len(picks_by_types[t]) > 0]
+
+    # let's take care of axis and figs
+    if axes is not None:
+        if not isinstance(axes, list):
+            axes = [axes]
+        from .utils import _validate_if_list_of_axes
```

Comment:
this shouldn't need to be nested, in fact there is already a `from .utils import ...` line at the top